api/src/main/java/io/grpc/ManagedChannelBuilder.java

```diff
@@ -55,8 +55,11 @@
    * </ul>
    *
    * <p>An authority string will be converted to a {@code NameResolver}-compliant URI, which has
-   * {@code "dns"} as the scheme, no authority, and the original authority string as its path after
-   * properly escaped. Example authority strings:
+   * the scheme from the name resolver with the highest priority (e.g. {@code "dns"}),
+   * no authority, and the original authority string as its path after properly escaped.
+   * We recommend specifying the schema explicitly if it is known, to avoid problems with other
```

Comment:
This is really working as designed. So I'm not wild about saying "problem." But I do agree it takes some coordination.

How about something like:

> We recommend libraries specify the schema explicitly if it is known, since libraries cannot know which NameResolver will be default during runtime.